modules/uhttp/client.go

```diff
@@ -0,0 +1,100 @@
+package uhttp
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"golang.org/x/net/context/ctxhttp"
+
+	"go.uber.org/fx/core"
+)
+
+// Client wraps around a http client
+type Client struct {
+	http.Client
+	filters []ClientFilter
+}
+
+// NewClient creates a new instance of uhttp Client
+func NewClient(client http.Client, filters ...ClientFilter) *Client {
+	if filters == nil {
```

Comment:
Basically, never check a slice for nil. A nil slice and an empty slice are different things, but functionaly should be treeated the same.